pkg/dashboard/resource/resource.go

```diff
@@ -90,14 +91,14 @@ func (r *resource) getDashboard() *dashboard.Server {
 	return r.GetServer().(*dashboard.Server)
 }
 
-func (r *resource) addAuthMiddleware() {
+func (r *resource) addAuthMiddleware(options auth.Options) {
 	authenticator := r.getDashboard().GetAuthenticator()
 	r.Logger.DebugWith("Installing auth middleware on router",
 		"authenticatorKind", authenticator.Kind(),
 		"resourceName", r.GetName())
-	r.GetRouter().Use(authenticator.Middleware())
+	r.GetRouter().Use(authenticator.Middleware(options))
 }
 
-func (r *resource) getCtxSession(request *http.Request) auth.Session {
-	return request.Context().Value(auth.ContextKeyByKind(r.getDashboard().GetAuthenticator().Kind())).(auth.Session)
+func (r *resource) getCtxSession(ctx context.Context) auth.Session {
```

Comment:
Suggestion:
Changing all occurrences of `getCtxSession` to get `ctx` does indeed make sense.
However this requires a lot of those occurrences to have and extra line of `ctx := request.Context()` where you didn't need it previously, so what if you have this function and also add:
```
func (r *resource) getRequestSession(request *http.Request) auth.Session {
	return r.getCtxSession(request.Context())
}
```
And this can be used in all the places where you have the request instance.
WDYT?

Reply:
I see your logic, however there is only 1 new line of `ctx := request.Context()`, all other resource methods already have a `ctx` which is being used for different things.
Also, I believe it's better to have a `ctx` var at your disposal for future usages (logs, k8s api etc).